### PR TITLE
new output for Centiloid pipeline

### PR DESCRIPTION
**Pull request by ariane-blk**
Added a CSV output to the Centiloid pipeline.

### CODE REVIEW

File: amypet/backend_centiloid.py [hunk 1]
```diff
@@ -392,6 +397,30 @@ def run(fpets, fmris, Cnt, tracer='pib', flip_pet=None, bias_corr=True,
...
 
+
+        # ---------------------------------
+        # > save csv with suvr and cl outputs
+        if csv_metrics == 'short':
+            csv_dict = {'path_outputs': out[onm]['opth'],
+                        **{f'suvr_{key}': value for key, value in out[onm]['suvr'].items()},
+                        **{f'cl_{key}': value for key, value in out[onm]['cl'].items() if key == 'wc'}}
+
+        if csv_metrics == 'long':
```
Comment (casperdcl) [inline amypet/backend_centiloid.py:408]:
```suggestion
        elif csv_metrics == 'long':
```

File: amypet/backend_centiloid.py [hunk 2]
```diff
@@ -392,6 +397,30 @@ def run(fpets, fmris, Cnt, tracer='pib', flip_pet=None, bias_corr=True,
...
+            csv_dict = {'path_outputs': out[onm]['opth'],
+                        **{f'suv_{key}': value for key, value in out[onm]['avgvoi'].items()},
+                        **{f'suvr_{key}': value for key, value in out[onm]['suvr'].items()},
+                        **{f'suvr_pib_calc_{key}': value for key, value in
+                           out[onm]['suvr_pib_calc'].items()},
+                        **{f'suvr_pib_calc_transf_{key}': value for key, value in
+                           out[onm]['suvr_pib_calc_transf'].items()},
+                        **{f'cl_{key}': value for key, value in out[onm]['cl'].items()}}
+
```
Comment (casperdcl) [inline amypet/backend_centiloid.py:417]:
```suggestion
        else:
            raise KeyError(f"`csv_metrics` must be 'short' or 'long' (got {csv_metrics})")
```

Or maybe

```suggestion
        elif csv_metrics:
            raise KeyError(f"`csv_metrics`: unknown value ({csv_metrics})")

        if csv_dict:
```

(so `False`/`None` will skip CSV outputs)

**Review by casperdcl — COMMENTED** [review]
minor suggestions

File: amypet/backend_centiloid.py (hunk 2 above)
Comment (ariane-blk) [reply amypet/backend_centiloid.py:417]: Yes better (probably option 2). @pjmark Do you have a preference? Could you please check this fits with the general format of your code?

**Review by pjmark — APPROVED** [review]

File: amypet/backend_centiloid.py [hunk 3]
```diff
@@ -392,6 +397,33 @@ def run(fpets, fmris, Cnt, tracer='pib', flip_pet=None, bias_corr=True,
...
+        elif csv_metrics:
+            raise KeyError(f"`csv_metrics`: unknown value ({csv_metrics})")
+
+        if csv_dict:
+        with open(os.path.join(outpath, 'amypet_outputs.csv'), 'w', newline='') as csvfile:
+            writer = csv.DictWriter(csvfile, fieldnames=csv_dict.keys())
+            writer.writeheader()
+            writer.writerow(csv_dict)
+        # ---------------------------------
+
```
Comment (casperdcl) [inline amypet/backend_centiloid.py:426]:
```suggestion
            with open(os.path.join(outpath, 'amypet_outputs.csv'), 'w', newline='') as csvfile:
                writer = csv.DictWriter(csvfile, fieldnames=csv_dict.keys())
                writer.writeheader()
                writer.writerow(csv_dict)
```

**Pull request merged**